plugins/ComposeAuth/src/commonMain/kotlin/io/github/jan/supabase/compose/auth/composable/GoogleAuth.kt

```diff
@@ -0,0 +1,7 @@
+package io.github.jan.supabase.compose.auth.composable
+
+import androidx.compose.runtime.Composable
+import io.github.jan.supabase.compose.auth.ComposeAuth
+
+@Composable
+expect fun ComposeAuth.rememberLoginWithGoogle(onResult: (NativeSignInResult) -> Unit = {}, fallback: () -> Unit = {}): NativeSignInState
```

Comment:
I think we could make this "fallback" lambda suspendable and default to `loginWithGoogle` ourselves, as the lambda is called from a LaunchedEffect. We should however add a `withContext(Dispatchers.IO)` around the fallback invoke.

Reply:
This should make it easy to implement other platforms for now, we could just use this:
```kotlin
val state = remember { NativeSignInState() }

LaunchedEffect(state.started) {
    if(state.started) {
         withContext(Disptachers.IO) {
               gotrue.loginWith(Google) //Or make a internal function for ComposeAuth like loginWithOAuth(provider)
               state.reset()
         }
    }
}
```

Reply:
Note for the other platforms, you only have to support JVM, JS and iOS for now as only these support Compose.

Reply:
Like the idea of having a default fallback and option for override. For other platforms can make force fallback logic for and make detailed implementation in other PRs ?

Reply:
Yes, I think that's the best idea for now